src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,173 @@
+import classNames from 'classnames';
+import {
+  ChangeEvent, useCallback, useEffect, useMemo, useRef, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader/Loader';
+import { Error } from '../../types/ErrorType';
+import { deleteTodos } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  todos: Todo[];
+  selectedTodosIds: number[];
+  errorNotification: string | null;
+  isAdding?: boolean;
+  handleOnChange: (id: number, data: Partial<Todo>) => Promise<void>;
+  setErrorNotification: React.Dispatch<React.SetStateAction<string | null>>;
+  setSelectedTodosIds: React.Dispatch<React.SetStateAction<number[]>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todos,
+  selectedTodosIds,
+  errorNotification,
+  isAdding,
+  handleOnChange,
+  setErrorNotification,
+  setSelectedTodosIds,
+  setTodos,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [selectedTodo, SetSelectedTodo] = useState(0);
+  const [newTodo, setNewTodo] = useState('');
+  const [doubleClick, setDoubleClick] = useState(false);
+
+  const { id, title, completed } = todo;
+
+  const isActive = useMemo(
+    () => selectedTodosIds.includes(id) || (isAdding && todo.id === 0),
+    [selectedTodosIds, isAdding, todo.id],
+  );
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [selectedTodo]);
+
+  const removeTodo = useCallback(async (TodoId: number) => {
+    setSelectedTodosIds([TodoId]);
+    try {
+      await deleteTodos(TodoId);
+
+      setTodos((prevTodos) => prevTodos.filter((item) => item.id !== TodoId));
+    } catch {
+      setErrorNotification(Error.Delete);
+    }
+  }, [todos, errorNotification]);
+
+  const UpdateTitle = useCallback(() => {
```

Comment:
Better name in one style, in my opinion, from the little letter. As you have with `removeTodo`.